src/Parallel/PhaseControl/PhaseChange.hpp

```diff
@@ -0,0 +1,212 @@
+enum ArbitrationStrategy {
+  /// Jump to the requested phase immediately, before considering other
+  /// requested phases.
+  ///
+  /// This will ensure that the requested phase is always run, where
+  /// alternative methods could have 'double-jumps' where the Main chare
+  /// replaces a requested phase immediately without actually entering the
+  /// phase.
+  RunPhaseImmediately,
+  /// After the requested phase is considered, continue considering other
+  /// requests, potentially replacing the this request.
+  ///
+  /// This will permit reprocessing the phase-jumps to help with cases where
+  /// multiple phases are simultaneously requested.
+  /// The `PermitAdditionalJumps` permits 'double-jumps' where a requested phase
+  /// is immediately replaced by another phase to jump to.
+  PermitAdditionalJumps
+};
+}
+
+/*!
+ * \brief `PhaseChange` objects determine the storage types and logic for
+ * moving between phases based on runtime data.
+ *
+ * There are two primary parts for the phase control flow:
+ *
+ * 1. On the side of the parallel components, they must select and/or compute
+ *    the runtime data necessary for choosing the next phase, then contribute it
+ *    to a global reduction to the Main component.
+ *    The components must then halt at a globally-valid state for the phase
+ *    change.
+ *    The requirements for the state will vary depending on the phase choices,
+ *    so triggers must be selected appropriately for the `PhaseChange` object.
+ *    For instance, selecting a common slab will usually represent a globally
+ *    well-behaved state for a `DgElementArray`.
+ * 2. On the side of the Main chare, the `PhaseChange` objects must use the
+ *    collected reduction data, or other persistent data stored in the Main
+ *    chare `phase_change_decision_data` to decide on a phase to request and
+ *    an `PhaseControl::ArbitrationStrategy` to determine how to resolve
+ *    multiple simultaneous requests.
+ *    Additionally, the `PhaseChange` objects must specify initialization
+ *    functions to set the starting state of the tags in
+ *    `phase_change_decision_data` for which they are responsible.
+ *
+ * In addition to the `static std::string name()` function, `options` type
+ * alias, and `static constexpr Options::String help()` function needed to be
+ * option-creatable, a derived class of `PhaseChange` must specify the
+ * type aliases:
+ * - `argument_tags`: A `tmpl::list` of  tags from the
+ *   \ref DataBoxGroup "DataBox" to be passed to `contribute_phase_data_impl` as
+ *   const references.
+ * - `return_tags`: A `tmpl::list` of mutable tags from the
+ *   \ref DataBoxGroup "DataBox" to be passed to `contribute_phase_data_impl` as
+ *   `gsl::not_null` pointers. This should be used only for tags that may be
+ *   altered during the `contribute_phase_data_impl` function.
+ * - `phase_change_tags_and_combines`: A `tmpl::list` of tags for populating the
```

Comment:
It is called `..._combines_list` in Main; this should be made consistent, but can be deferred to a clean-up PR if you prefer